[Feat #22] 회원 서비스 구현

JoHyungJun opened the pull request:
### 변경사항
<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요. 가급적이면 as-is, to-be를 활용해서 작성해주세요.  -->
**AS-IS**
- 회원이 이용할 기초적인 서비스 구현 필요성

**TO-BE**
- 비밀번호 수정 기능 구현
- 회원 탈퇴 기능 구현 (soft delete 으로 status 값을 변경)
- 토큰 재발급 기능 구현 (redis 와 연동하여 refresh 토큰을 redis 에 저장하고 관리)

### 테스트
<!-- 본 변경사항이 테스트가 되었는지 기술해주세요 --> 
- [ ] 테스트 코드
- [X] API 테스트

JoHyungJun committed:
[Feat #22] redis 기본 설정 및 member service
- redis, redisson 기본 설정
- 비밀번호 변경, 회원 탈퇴

JoHyungJun committed:
[Feature #22] cors 설정

JoHyungJun committed:
[Feature #22] 회원 서비스 전체 구현
- 토큰 재발급
- 비밀번호 수정
- 회원 탈퇴

ParkHaeBeen commented:
코드는 좋은데 지금 build쪽에서 실패가 떠서 보니까 openapi관련해서 없어서 오류가 뜨느 거같아요
![image](https://github.com/ZB-Keyword/Keyword-BackEnd/assets/130157565/e04e53e7-1f2f-44c4-bd71-1f20e7cf809f)

이거 실행주셔서 static/docs에 open관련한 json한번 만들어주시면 해결될 수도 있을거같아요.

JoHyungJun committed:
[Feature #22] member 서비스 수정
- token response 에 해당 member 의 정보 함께 반환
- phone 에 unique 설정 추가 및 회원 가입 시 validate

ParkHaeBeen committed:
Merge branch 'develop' into feature/#22-redis-memberservice
